conans/client/manager.py

```diff
@@ -341,22 +291,23 @@ def install(self, reference, current_path, profile, remote=None,
             #           "other_tool/3.2@lasote/stable": (deps_cpp_info, deps_env_info)}
             #  "None": {"cmake/3.1@lasote/stable": (deps_cpp_info, deps_env_info)}
             build_dep_infos = self._install_build_requires_and_get_infos(profile, loader, remote_proxy, current_path)
+            loader.initial_deps_infos = build_dep_infos
 
         # Build the graph for the real dependency tree
-        conanfile, is_txt = self._get_conanfile_object(loader, reference, filename, current_path)
-        graph_builder = self._get_graph_builder(loader, update, remote_proxy, build_dep_infos)
+        conanfile = self._get_conanfile_object(loader, reference, filename, current_path)
+        graph_builder = self._get_graph_builder(loader, update, remote_proxy)
         deps_graph = graph_builder.load(conanfile)
 
         # This line is so the conaninfo stores the correct complete info
-        self._fill_info(conanfile, is_txt)
+        conanfile.info.scope = profile.scopes
```

Comment:
Sure, but able to remove the other lines of _fill_info, now it is only about setting the scopes for storing them in conaninfo.txt. Still bad, but slightly better than previously.